### ISSUE TITLE
Unable to use Windows SSL trust store

### ISSUE THREAD

**Issue by knolleCC** [gradle/gradle#6584]
The JRE can be set to use the Windows trust store via `javax.net.ssl.trustStoreType=Windows-ROOT`.
Gradle seems to have its own trust store handling which does not respect this setting.
As far as I understand the code in `DefaultSslContextFactory.java`, there always needs to be a trust store file (`javax.net.ssl.trustStore=SOME-FILE`) or code will fall back to use the default JRE files.
Why does gradle not use the Java default method?

Actual problem:
I want to use a maven repository via HTTPS with a special company certificate.

Goal: 
Support Windows trust store usage via `javax.net.ssl.trustStoreType=Windows-ROOT/MY` as the JRE does.

**Comment by wendlm** [comment]
We encounter the same problem which is very annoying actually, cause we have to add our own company certificate to the JVM keystore each time we change the JVM.

**Comment by hbdesiato** [comment]
There is a workaorund: Just set `javax.net.ssl.trustStore` to any readable regular file. This file will be ignored if you set `javax.net.ssl.trustStoreType=Windows-ROOT`.

Adding the following lines to your gradle.properties file should work:
```
systemProp.javax.net.ssl.trustStore=C:\\Windows\\win.ini
systemProp.javax.net.ssl.trustStoreType=Windows-ROOT

```

**Comment by TTMaZa** [comment]
Have you tried 
systemProp.javax.net.ssl.trustStore=NUL

**Comment by justin-michel-boeing** [comment]
Setting the trustStore property to the win.ini worked for me, but others on my team now can't build and are getting  IOException:  Invalid keystore format 

Is anyone acknowledging that this is a bug in gradle, and are there plans to fix it?

**Comment by knolleCC** [comment]
> Adding the following lines to your gradle.properties file should work:
> 
> ```
> systemProp.javax.net.ssl.trustStore=C:\\Windows\\win.ini
> systemProp.javax.net.ssl.trustStoreType=Windows-ROOT
> ```

Thanks @hbdesiato, this works for me. Still I would appreciate this bug to be fixed and still I wonder why they do not use the default JRE mechanism.

**Comment by knolleCC** [comment]
> Still I would appreciate this bug to be fixed and still I wonder why they do not use the default JRE mechanism.

Uhm, seems I have to apologize. Using Gradle 6.0.1, just specifying "javax.net.ssl.trustStoreType=Windows-ROOT" now works for me. Did they fix this already?

**Comment by knolleCC** [comment]
> Uhm, seems I have to apologize. Using Gradle 6.0.1, just specifying "javax.net.ssl.trustStoreType=Windows-ROOT" now works for me. Did they fix this already?

Hmm, forget what I posted in Jan... it still only works with a file path (trustStore parameter). The code prooves it: https://github.com/gradle/gradle/blob/master/subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/DefaultSslContextFactory.java

**Comment by MarkRx** [comment]
This still happens in Gradle 7.4.2. Please re-open.

**Comment by guai** [comment]
@wolfs pls reopen

**Comment by jbartok** [comment]
The issue is in the backlog of the relevant team, but the existence of a workaround makes it non-critical, so it might take a while before a fix is made.

**Comment by ov7a** [comment]
Fixed by #25106
